vignettes/mmrm_review_methods.Rmd

```diff
@@ -0,0 +1,277 @@
+</code></pre>
+
+## Compound symmetry (homogeneous)
+
+### SAS
+<pre><code>PROC MIXED DATA = fev_data cl method=ml;
+CLASS AVISIT(ref = 'VIS4') USUBJID;
+MODEL FEV1 = / ddfm=satterthwaite solution chisq;
+<b>REPEATED AVISIT / subject=USUBJID type=CS</b> r=1, 2, 3, 4, 5, 6, 7, 8, 9, 10 rcorr;
+</code></pre>
+
+### mmrm
+<pre><code>mmrm::mmrm(formula = FEV1 ~ <b>cs(VISITN | USUBJID)</b>, 
+  data = fev_data, reml = FALSE)
+</code></pre>
+
+## Auto-regressive (heterogeneous)
+
+### SAS
+<pre><code>PROC MIXED DATA = fev_data cl method=ml;
+CLASS AVISIT(ref = 'VIS4') USUBJID;
+MODEL FEV1 = / ddfm=satterthwaite solution chisq;
+<b>REPEATED AVISIT / subject=USUBJID type=ARH(1)</b> r=1, 2, 3, 4, 5, 6, 7, 8, 9, 10 rcorr;
+</code></pre>
+
+### mmrm
+<pre><code>mmrm::mmrm(formula = FEV1 ~ <b>ar1h(VISITN | USUBJID)</b>, 
+  data = fev_data, reml = FALSE)
+</code></pre>
+
+### nlme::gls
+<pre><code>nlme::gls(formula = FEV1 ~  SEX + ARMCD * AVISIT,
+  data = fev_data,
+  correlation = <b>nlme::corCAR1(form = ~AVISIT | USUBJID)</b>,
+  weights = nlme::varIdent(form = ~1|AVISIT),
+  na.action = na.omit)
+</code></pre>
+
+## Auto-regressive (homogeneous)
+
+### SAS
+<pre><code>PROC MIXED DATA = fev_data cl method=ml;
+CLASS AVISIT(ref = 'VIS4') USUBJID;
+MODEL FEV1 = / ddfm=satterthwaite solution chisq;
+<b>REPEATED AVISIT / subject=USUBJID type=AR(1)</b> r=1, 2, 3, 4, 5, 6, 7, 8, 9, 10 rcorr;
+</code></pre>
+
+### mmrm
+<pre><code>mmrm::mmrm(formula = FEV1 ~ <b>ar1(VISITN | USUBJID)</b>, 
+  data = fev_data, reml = FALSE)
+</code></pre>
+
+## Ante-dependence (heterogeneous)
+
+### SAS
+<pre><code>PROC MIXED DATA = fev_data cl method=ml;
+CLASS AVISIT(ref = 'VIS4') USUBJID;
+MODEL FEV1 = / ddfm=satterthwaite solution chisq;
+<b>REPEATED AVISIT / subject=USUBJID type=ANTE(1)</b> r=1, 2, 3, 4, 5, 6, 7, 8, 9, 10 rcorr;
+</code></pre>
+
+### mmrm
+<pre><code>mmrm::mmrm(formula = FEV1 ~ <b>adh(VISITN | USUBJID)</b>, 
+  data = fev_data, reml = FALSE)
+</code></pre>
+
+## Ante-dependence (homogeneous)
+
+### mmrm
+<pre><code>mmrm::mmrm(formula = FEV1 ~ <b>ad(VISITN | USUBJID)</b>, 
+  data = fev_data, reml = FALSE)
+</code></pre>
+
+# Benchmarking
+```{r, cache=TRUE, warning=FALSE, message=FALSE}
+library(mmrm)
+library(lme4)
+library(glmmTMB)
+
+mb <- microbenchmark::microbenchmark(times = 10,
+  fit_gls = nlme::gls(FEV1 ~ RACE + ARMCD * AVISIT, data = fev_data, 
+    na.action = na.omit,
+    correlation = nlme::corSymm(form = ~ VISITN | USUBJID),
+    weights = nlme::varIdent(form = ~ 1|VISITN)),
+  fit_mmrm = mmrm::mmrm(
+    formula = FEV1 ~ RACE + ARMCD * AVISIT + us(AVISIT | USUBJID),
+    data = fev_data
+  ),
+  fit_lmer = lme4::lmer(
+    FEV1 ~ RACE + ARMCD * AVISIT + (0 + AVISIT | USUBJID),
+    data = fev_data, control = lmerControl(check.nobs.vs.nRE = "ignore"),
+    na.action = na.omit),
+  fit_glmmtmb = glmmTMB::glmmTMB(
+    FEV1 ~ RACE + ARMCD * AVISIT + us(0 + AVISIT | USUBJID),
+    data = fev_data, dispformula = ~ 0, REML = TRUE)
+ )
+mb
```

Comment:
I think the current structure is a good start!
Some ideas for next steps:
1) can we have a table to show which software supports which covariance structures (restricting to those the `mmrm` package has e.g. since SAS has too many to show all)
2) can we have graphs on a) the run time needed and b) comparison of results to SAS e.g. in terms of relative differences?

Reply:
cool, thanks for this feedback and giving me a direction for where to take this next. I'll add some check boxes up in the issue description for these.